tasks/Module 3/task_2_test.py

```diff
@@ -0,0 +1,47 @@
+"""Task 2 of Module 3."""
+
+import pytest
+
+
+def my_splitter(to_split, separator=None):
+    """Create personal str.split() implementation with UnitTest."""
+    if separator is None:
+        raise TypeError(
+            "YOU SHALL NOT PASS..unless you'll provide a separator")
```

Comment:
I like the error message, however the task was to:
> implement my_split function that mimics built in split() method as close as possible.

Please take a look at docs for `split` method of `str` type:
```
>>> help(str.split)
```

Reply:
error has been replaced with findall method of regex.

@musk-ox @Pacu2 Guys, could you please advice, if we could(somehow) use our `separator` argument as an input for regex? I know we should use raw strings for it, but maybe you know a way? :)